tests/controller.test.ts

```diff
@@ -0,0 +1,860 @@
+import request from 'supertest';
+import createApplication from '../src/createApp';
+import { FindGroupError } from '../src/customErrors';
+import { User } from '../src/models/typeORMModels';
+
+const mockUserService = {
+  createUser: jest.fn(),
+  getUserById: jest.fn(),
+  updateUser: jest.fn(),
+  deleteUser: jest.fn(),
+  getUsers: jest.fn(),
+  getUserByCredentials: jest.fn(),
+};
+
+const mockGroupService = {
+  createGroup: jest.fn(),
+  getGroupById: jest.fn(),
+  updateGroup: jest.fn(),
+  deleteGroup: jest.fn(),
+  getGroups: jest.fn(),
+  addUsersToGroup: jest.fn(),
+};
+
+const validCredentials = {
+  login: 'log123',
+  password: 'password1234',
+};
+
+const validUserData = {
+  ...validCredentials,
+  age: 30,
+  is_deleted: false,
+};
+
+const validGroupName = 'grpup1';
+const notValidGroupName = 'g';
+const validGroupId = '44e042a2-2acd-466f-bdfc-ec1e7a671dsd';
+const notValidGroupId = 'not valid ID';
+const validGroupPermissions = [
+  'READ',
+  'WRITE',
+  'DELETE',
+  'SHARE',
+  'UPLOAD_FILES',
+];
+const notValidGroupPermissions = [
+  'READ1',
+  'WRITE',
+  'DELETE',
+  'SHARE',
+  'UPLOAD_FILES',
+];
+const notValidJWT = 'not valid string';
+const validUserId = '44e042a2-2acd-466f-bdfc-ec1e7a671dsd';
+const validUserIds = [
+  '44e042a2-2acd-466f-bdfc-ec1e7a671dsd',
+  '54e042a2-2acd-466f-bdfc-ec1e7a671dsd',
+];
+const notValidUserIds = [
+  '44e042a2-2acd-466f-bdfc-ec1e7a671d',
+  '54e042a2-2acd-466f-bdfc-ec1e7a671d',
+];
+const notValidLogin = 'a';
+const notValidPassword = '1';
+const notValidUserId = 'not valid ID';
+const validSubstring = 'log';
+const notValidSubstring = 'logaaaaaaaaaaa';
+const validLimit = 3;
+const notValidLimit = 'a';
+
+const validGroupData = {
+  name: validGroupName,
+  permissions: validGroupPermissions,
+};
+
+async function getValidJWT() {
+  const credentials = validCredentials;
+  const user = new User();
+  mockUserService.getUserByCredentials.mockReturnValue(user);
+  const response = await request(application)
+    .post('/user/login')
+    .send(credentials);
+  return response.body.token;
+}
+
+const { app: application, server } = createApplication(
+  mockUserService,
+  mockGroupService
+);
+
+describe('CONTROLLER: ', () => {
+  afterAll(() => {
+    server.close();
+  });
+
+  describe('User Controller: ', () => {
+    describe('POST /user/login', () => {
+      beforeEach(() => {
+        mockUserService.getUserByCredentials.mockClear();
+      });
+      describe('when credentials passed validation', () => {
+        test('should trigger getUserByCredentials method with credentials', async () => {
+          const body = validCredentials;
+          await request(application).post('/user/login').send(body);
```

Comment:
```suggestion
         const response = await request(application).post('/user/login').send(body);
```
And then check that response.body has expected value. Add this to all your tests, please.

Reply:
response don't used in this case

Reply:
You need to check body as well, to be sure it complies with the contract. So if anyone changes it by accident, test will fail.